### DISCUSSION TITLE
Relay+Reactive support

### DISCUSSION THREAD

**Discussion by ruanfi** [kobylynskyi/graphql-java-codegen#482] [Ideas]
Bogdan, hi, I need generic over generic responses for my schema. Reactive Mono<T> over relay's Connection<T> So I should provided something like this:
`relayConfig.connectionType = "reactor.core.publisher.Mono<graphql.relay.Connection<????>>"`
But it doesn't work with generic inside generic.
I checked a code, and find the `getGenericsStringmethod` inside `JavaGraphQLTypeMapper.java`.
I changed it a bit:
```java
    @Override
    public String getGenericsString(MappingContext mappingContext, String genericType, String typeParameter) {
        if (genericType.contains("%s")) {
            return String.format(genericType, typeParameter);
        } else {
            return String.format("%s<%s>", genericType, typeParameter);
        }
    }
```
So now I can provide a config variable like this:
`relayConfig.connectionType = "reactor.core.publisher.Mono<graphql.relay.Connection<%s>>"`
And I got a code I need.
Is there a way to achieve the same without generator code changes, and if there is no other way, can you consider such a change inside a master dev?
Thanks.

**Comment by kobylynskyi** [comment]
@ruanfi this code change looks ok to me as it will not break the existing functionality and will provide a more flexible approach.
I've applied the change in #485

**Reply by kobylynskyi** [reply]
Will release a new version soon.

**Reply by ruanfi** [reply]
Thanks a lot. (:
The same patch could be applied to scala and kotlin parts as well (:

**Reply by ruanfi** [reply]
Bogdan, hi, do you have a plans to release this changes?

**Reply by kobylynskyi** [reply]
@ruanfi, I released 4.1.2 today.

**Reply by ruanfi** [reply]
Bogdan, thanks a lot! ((: 👍
